llvm/lib/Target/RISCV/RISCVISelLowering.h

```diff
@@ -977,19 +979,60 @@ class RISCVTargetLowering : public TargetLowering {
   unsigned getMinimumJumpTableEntries() const override;
 };
 
+class RVVArgDispatcher {
+public:
+  static constexpr unsigned NumArgVRs = 16;
```

Comment:
It would be nice if you found a way to include some of the following verbiage into this patch in the relevant areas of the patch, so a reader can have a better understanding of the context.
```
The rules for passing vector arguments are as follows:

1. For the first vector mask argument, use v0 to pass it.
2. For vector data arguments or rest vector mask arguments, starting from the v8 register, if a vector register group between v8-v23 that has not been
allocated can be found and the first register number is a multiple of LMUL,
then allocate this vector register group to the argument and mark these
registers as allocated. Otherwise, pass it by reference and are replaced in
the argument list with the address.
3. For tuple vector data arguments, starting from the v8 register, if NFIELDS
consecutive vector register groups between v8-v23 that have not been allocated
can be found and the first register number is a multiple of LMUL, then allocate
these vector register groups to the argument and mark these registers as
allocated. Otherwise, pass it by reference and are replaced in the argument list
with the address.
```

For example, here might be a good place to explain why there are 16 registers available (v0 and v8-v23).